content/articles/css-supports-query/index.md

```diff
@@ -0,0 +1,244 @@
+      tincidunt augue. Integer malesuada nunc vel risus commodo viverra maecenas
+      accumsan lacus. Tempor orci eu lobortis elementum. Placerat duis ultricies
+      lacus sed. Adipiscing enim eu turpis egestas pretium aenean pharetra
+      magna. In massa tempor nec feugiat nisl. Diam maecenas sed enim ut sem
+      viverra aliquet eget. Lacinia at quis risus sed vulputate odio. Aliquet
+      lectus proin nibh nisl condimentum id venenatis. Id venenatis a
+      condimentum vitae sapien pellentesque. Arcu vitae elementum curabitur
+      vitae nunc. Non enim praesent elementum facilisis leo vel fringilla. Nisl
+      pretium fusce id velit ut.
+    </p>
+  </body>
+</html>
+```
+
+Open this file on a modern browser. Let us use [Chrome version 90](https://www.google.com/intl/en_in/chrome/). You would notice that the hero banner sticks to the top. Try scrolling to the bottom, and you would notice that the hero banner still sticks to the top. Therefore, we can confirm that the `position: sticky` feature works.
+
+![Position Sticky](/engineering-education/css-supports-query/sticky.jpg)
+
+*Position Sticky on Chrome 90*
+
+Now, download and install [Firefox version 22](https://ftp.mozilla.org/pub/firefox/releases/22.0/). Open `index.html` in Firefox. You will notice that the hero banner does not scroll. The `position: sticky` feature does not work.
+
+![Position Sticky Firefox](/engineering-education/css-supports-query/sticky-firefox.jpg)
+
+*Position Sticky on Firefox 22*
+
+Let us now fix this using `@supports`.
+
+**`index.html`**
+
+```html
+<style>
+  body {
+    margin: 0;
+  }
+  .hero {
+    /* Sets the background color of the hero banner */
+    background-color: blue;
+    /* Sets padding on all sides */
+    padding: 2rem;
+    /* Sticky position sticks and scrolls along with the page */
+    position: sticky;
+    width: 100%;
+    height: 100px;
+    /* Top property is used to ensure that the hero banner is on the top */
+    top: 0;
+  }
+
+  /* If the browser does not support the position: sticky property, the rules within this block are applied */
+  @supports not (position: sticky) {
+    .hero {
+      /* We set the position to be fixed so that the hero banner stays at the top */
+      position: fixed;
+    }
+    body {
+      /* We set the top margin to be 100px so that the hero banner doesn't overlap the placeholder text */
+      margin-top: 100px;
+    }
+  }
+  .text {
+    /* Sets the font size */
+    font-size: x-large;
+  }
+</style>
+```
+
+If you run it on Firefox 22, you would notice that the hero banner sticks to the top, even when we scroll to the bottom of the page.
+
+In the above example, we use the `@supports` feature query and the **not** operator to check for the browsers that do not support the `position: sticky` CSS rule. We specify some alternative CSS rules for older browsers within the `@supports` block so that the CSS doesn't break in older browsers.
```

Comment:
You only talked about the "not" operation. Please talk about "or" and "and" operations. Create 3 headers and describe the differences between them and when to use them.